docs/user-guide/multicontract/index.md

```diff
@@ -0,0 +1,675 @@
+
+The "Call Resolution" tab on the report provides more information about all of
+the unresolved method calls within the contract and how they are resolved by
+the Prover[^resolutionWarnings]:
+
+![Call resolution for `integrityOfDeposit` showing havocs of return values for `balanceOf` and all variables of external contracts for `transferFrom`](no-link-call-resolution.png)
+
+[^resolutionWarnings]: Unresolved calls that are not explicitly handled are
+  considered warnings; in this case there are three unresolved calls, which is
+  why there is a red 3 on the call resolution tab.  In general, it is good
+  practice to explicitly resolve all calls.
+
+Here we see that the call from `Pool.deposit` to `balanceOf` is summarized by
+havocing only the return value (since `balanceOf` is a view method), while the
+call from `Pool.deposit` to `transferFrom` havocs all contracts except `Pool`.
+
+Working with known contracts
+----------------------------
+
+In the case of the `Pool` verification, we don't want the Prover to choose
+arbitrary behavior for the `Asset`, because we have the `Asset` code.  Instead,
+we would like the Prover to model the `asset` contract using the `Asset` code.
+
+To do so, we must first add the `Asset` contract to the set of contracts that
+the Prover knows about.  This set of contracts is called the {term}`scene`.  You
+can add a contract to the scene by passing the solidity source as a
+[command line argument](/docs/ref-manual/cli/options.md)
+to `certoraRun`.  The Prover creates a contract instance (with a corresponding
+address) in the scene for each source contract provided on the command line:
+
+```sh
+$ certoraRun contracts/Pool.sol contracts/Asset.sol --verify Pool:certora/specs/pool_havoc.spec ...
+```
+
+Adding `Asset.sol` to the scene makes the Prover aware of it, but it does not
+connect the `asset` field of the pool to the `Asset` contract.  Although
+`Pool.asset` is declared to have type `Asset` in the solidity source, the
+solidity compiler erases that information from the bytecode; in the compiled
+bytecode the field is just treated as an `address`.
+
+To reconnect the `Asset` code to the `Pool.asset` field, we can use the
+{ref}`--link` option ([full script][pool-link-script]):
+
+```sh
+$ certoraRun contracts/Pool.sol contracts/Asset.sol \
+    --verify Pool:certora/specs/pool_havoc.spec \
+    --link   Pool:asset=Asset \
+    ...
+```
+
+The `--link Pool:asset=Asset` option tells the Prover to assume that the `asset`
+field of the `Pool` contract instance in the scene is a pointer to the `Asset`
+contract instance in the scene.  With this information, the Prover is able to
+resolve the calls to the methods on `Pool.asset` using the code in `Asset.sol`.
+
+With this option, the Prover is no longer able to construct a counterexample to
+the `integrityOfDeposit` rule, so the rule passes.
+
```

Comment:
should we say something about the call resolution - it does not apper there any more. right? ( very confusing what is in and is not in the call resolution) 
We can also give links to result page so one can see how it should look

Reply:
Updated.  I considered adding run links, but I think I asked and it wasn't clear if these are persistent enough to include in the documentation.  Even if we do, I would want to have more process in place to make sure we are rebuilding the examples when they change.